File: src/main/java/net/gegy1000/terrarium/server/map/glob/generator/VegetationWithCropland.java
package net.gegy1000.terrarium.server.map.glob.generator;

import net.gegy1000.terrarium.server.map.glob.GlobType;
import net.minecraft.world.gen.layer.GenLayer;
import net.minecraft.world.gen.layer.GenLayerFuzzyZoom;
import net.minecraft.world.gen.layer.GenLayerVoronoiZoom;

public class VegetationWithCropland extends MultiGlobGenerator {
    public VegetationWithCropland() {
        super(GlobType.VEGETATION_WITH_CROPLAND, new Entry(GlobType.IRRIGATED_CROPS, 30),
                new Entry(GlobType.GRASSLAND, 25),
                new Entry(GlobType.SHRUBLAND, 25),
                new Entry(GlobType.FOREST_SHRUBLAND_WITH_GRASS, 20));
    }

    @Override
    protected GenLayer zoom(GenLayer layer) {
        GenLayer zoom = new GenLayerVoronoiZoom(1000, layer);
        zoom = new GenLayerFuzzyZoom(2000, zoom);
        zoom = new GenLayerVoronoiZoom(3000, zoom);
        return zoom;
    }
}

File: src/main/kotlin/net/gegy1000/terrarium/server/map/glob/generator/VegetationWithCropland.kt
package net.gegy1000.terrarium.server.map.glob.generator

import net.gegy1000.terrarium.server.map.glob.GlobType
import net.minecraft.world.gen.layer.GenLayer
import net.minecraft.world.gen.layer.GenLayerFuzzyZoom
import net.minecraft.world.gen.layer.GenLayerVoronoiZoom

class VegetationWithCropland : MultiGlobGenerator(GlobType.VEGETATION_WITH_CROPLAND,
        Entry(GlobType.IRRIGATED_CROPS, 30),
        Entry(GlobType.GRASSLAND, 25),
        Entry(GlobType.SHRUBLAND, 25),
        Entry(GlobType.FOREST_SHRUBLAND_WITH_GRASS, 20)) {

    override fun zoom(layer: GenLayer): GenLayer {
        var zoom: GenLayer = GenLayerVoronoiZoom(1000, layer)
        zoom = GenLayerFuzzyZoom(2000, zoom)
        zoom = GenLayerVoronoiZoom(3000, zoom)
        return zoom
    }
}
